lib/datadog/open_feature/binding/configuration.rb

```diff
@@ -0,0 +1,292 @@
+# frozen_string_literal: true
+
+module Datadog
+  module OpenFeature
+    module Binding
+      # Variation types supported by UFC
+      module VariationType
+        STRING = 'STRING'
+        INTEGER = 'INTEGER'
+        NUMERIC = 'NUMERIC'
+        BOOLEAN = 'BOOLEAN'
+        JSON = 'JSON'
+      end
+
+      # Condition operators for rule evaluation
+      module ConditionOperator
+        MATCHES = 'MATCHES'
+        NOT_MATCHES = 'NOT_MATCHES'
+        GTE = 'GTE'
+        GT = 'GT'
+        LTE = 'LTE'
+        LT = 'LT'
+        ONE_OF = 'ONE_OF'
+        NOT_ONE_OF = 'NOT_ONE_OF'
+        IS_NULL = 'IS_NULL'
+      end
+
+      # Assignment reasons returned in ResolutionDetails
+      module AssignmentReason
+        TARGETING_MATCH = 'TARGETING_MATCH'
+        SPLIT = 'SPLIT'
+        STATIC = 'STATIC'
+      end
+
+      # Represents a feature flag configuration
+      class Flag
+        attr_reader :key, :enabled, :variation_type, :variations, :allocations
+
+        def initialize(key:, enabled:, variation_type:, variations:, allocations:)
+          @key = key
+          @enabled = enabled
+          @variation_type = variation_type
+          @variations = variations || {}
+          @allocations = allocations || []
+        end
+
+        def self.from_json(key, flag_data)
+          new(
+            key: key,
+            enabled: flag_data['enabled'] || false,
+            variation_type: flag_data['variationType'],
+            variations: parse_variations(flag_data['variations'] || {}),
+            allocations: parse_allocations(flag_data['allocations'] || [])
+          )
+        end
+
+        private
+
+        def self.parse_variations(variations_data)
+          variations_data.transform_values do |variation_data|
+            Variation.from_json(variation_data)
+          end
+        end
+
+        def self.parse_allocations(allocations_data)
+          allocations_data.map { |allocation_data| Allocation.from_json(allocation_data) }
+        end
+      end
+
+      # Represents a variation value for a flag
+      class Variation
+        attr_reader :key, :value
+
+        def initialize(key:, value:)
+          @key = key
+          @value = value
+        end
+
+        def self.from_json(variation_data)
+          new(
+            key: variation_data['key'],
+            value: variation_data['value']
+          )
+        end
+      end
+
+      # Represents an allocation rule with traffic splits
+      class Allocation
```

Comment:
does it make sense to name this class `AllocationRule`, if it represents a rule?